x-pack/legacy/plugins/maps/public/embeddable/map_embeddable.tsx

```diff
@@ -42,11 +56,60 @@ import {
 } from '../../../../../plugins/maps/public/reducers/non_serializable_instances';
 import { getMapCenter, getMapZoom, getHiddenLayerIds } from '../selectors/map_selectors';
 import { MAP_SAVED_OBJECT_TYPE } from '../../common/constants';
+import { RenderTooltipContentParams } from './types';
+
+interface MapConfig {
+  editUrl?: string;
+  indexPatterns: IIndexPattern[];
+  editable: boolean;
+  title?: string;
+  layerList: unknown[];
+}
+
+export interface MapEmbeddableInput extends EmbeddableInput {
+  timeRange?: TimeRange;
```

Comment:
Something we should probably consider, but the idea is that not all containers will pass those values down and not all embeddables will need those values (e.g. a Markdown element maybe).   But they are such primary concepts, and as long as they are optional, it might be nicer to move it.

Same with IndexPattern output.  There really is only one prod container - dashboard.  Canvas is the other one I was thinking of, but bc of their unique use case (no global query bar, for instance), they aren't and don't have a motivation to use the container interface.

So yea, good thought, I think we'll just wait and see for now, come back to this later.